flake.nix

```diff
@@ -0,0 +1,75 @@
+{
+  description = "A very basic flake";
+
+  inputs = {
+    # latex indent is broken on the latest nixpkgs : /
+    nixpkgs.url = "github:nixos/nixpkgs?rev=ef7226d68ba45b2de3e428e5d4bb4532caffec7b";
+    flake-parts.url = "github:hercules-ci/flake-parts";
+    pre-commit-hooks.url = "github:cachix/pre-commit-hooks.nix";
+  };
+
+  outputs = inputs@{ flake-parts, nixpkgs, pre-commit-hooks, ... }:
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      imports = [
+        inputs.pre-commit-hooks.flakeModule
+      ];
+
+      systems = [ "x86_64-linux" ];
+
+      perSystem = { system, config, ... }:
+        let
+          pkgs = import nixpkgs {
+            inherit system;
+          };
+
+          # Ideally we figure out what exact package is used, but too much work
+          tex = pkgs.texlive.combine { inherit (pkgs.texlive) scheme-full; };
```

Comment:
This is less of an issue since github CI can cache derivation built. So this is only heavy on the first time building and storage-wise.

Reply:
Holy moly thanks a lot for the feedback!

let me fix

Reply:
https://gist.github.com/SeungheonOh/4b5ab2d4815f754d8d9cd473f4d3335e

Here's what I played with, I didn't bother fighting out all the deps though

Reply:
@SeungheonOh just pushed this https://github.com/elRaulito/eUTxO-Fundamentals-Building-Cardano-Smart-Contracts

That fixes all broken dependencies

Is it conflincting with the PR?

Happy holidays btw and thanks for your help